### ISSUE TITLE
Incorrect enum value of KeyModifierMask.MODIFIER_MASK

### ISSUE THREAD

**Issue by l0g0sys** [godotengine/godot#89775]
### Tested versions

- Reproducible in: master branch

### System information

All

### Issue description

File core/os/keyboard.h, line 252 contains incorrect value assignment:
`MODIFIER_MASK = (0x7F << 22), ///< Apply this mask to isolate modifiers.`

Correct value assignment is:
`MODIFIER_MASK = (0x7E << 22)`

### Explanation
22nd bit is part of Key enum in form of Key.SPECIAL value and not KeyModifierMask enum, as no value in KeyModifierMask enum contains 22nd bit set.

In other words `KeyModifierMask.CODE_MASK & KeyModifierMask.MODIFIER_MASK` must evaluate to zero.

### Minimal reproduction project (MRP)

None
